### PR TITLE
Update to Sodium 0.5.4/0.5.5 and Minecraft 1.20.3/1.20.4

### PR DESCRIPTION
**Pull request by Vieur**
This is my first time porting this mod, so I might be unable to fix it if you guys encounter any issues.

EDIT: I saw that 1.20.4 has been released. This should still work on that version.

### CODE REVIEW

**Comment by Vieur** [comment]
Download the compiled version here!
https://github.com/Vieur/Indium/releases/tag/1.20.3

**Comment by Vaskyy** [comment]
works pretty good :) thanks for that!
--------------------
without shaders
without shaders
--------------------
with shaders
with shaders

**Review by JonasFranke — APPROVED** [review]

**Comment by sternschnaube** [comment]
Thanks for the update. Is it still compatible with 1.20.2?

**Comment by D-argon** [comment]
Works with Sodium .5.5 too, no issues found yet

> Thanks for the update. Is it still compatible with 1.20.2?

I had no problem with it

**Comment by sternschnaube** [comment]
Thanks, that's perfect 🥰

**Comment by Eutropios** [comment]
Hi there. The lack of an Indium update that allows for a newer version of Sodium is blocking my ability to update a handful of mods that rely on a higher version of Sodium and Iris. Is this update planned to be released soon, or should I remove Indium from my instance for now?

**Comment by sternschnaube** [comment]
Calm, and stay on the Sodium version which works with your setup for now.

You know the answer already, this comment won't bring the official update faster out.

**Comment by D-argon** [comment]
> Hi there. The lack of an Indium update that allows for a newer version of Sodium is blocking my ability to update a handful of mods that rely on a higher version of Sodium and Iris. Is this update planned to be released soon, or should I remove Indium from my instance while I wait for a maintainer to approve the workflow?

1. you don't need the merge/official release to download this update that is currently supporting the newest version of Sodium;

2. if you're not helping/developing the mod you shouldn't complain about lack of update or rush one;

  so if it's crucial for you to have an official release, you should keep the other dependents downgraded and, as you said, wait, or find other alternatives.

**Comment by ZincHere** [comment]
> Download the compiled version here! https://github.com/Vieur/Indium/releases/tag/1.20.3

Very cool! But my game keeps crashing :'c (I use a bunch of other mods, so that might be why, but is there a way you can fix this?)

**Comment by sternschnaube** [comment]
> > Download the compiled version here! https://github.com/Vieur/Indium/releases/tag/1.20.3
> 
> Very cool! But my game keeps crashing :'c (I use a bunch of other mods, so that might be why, but is there a way you can fix this?)

Give us more information >>> provide us your crash log. We can't work with air.

**Comment by Eutropios** [comment]
> > Hi there. The lack of an Indium update that allows for a newer version of Sodium is blocking my ability to update a handful of mods that rely on a higher version of Sodium and Iris. Is this update planned to be released soon, or should I remove Indium from my instance while I wait for a maintainer to approve the workflow?
> 
>     1. you don't need the merge/official release to download this update that is currently supporting the newest version of Sodium;
> 
>     2. if you're not helping/developing the mod you shouldn't complain about lack of update or rush one;
> 
> 
> so if it's crucial for you to have an official release, you should keep the other dependents downgraded and, as you said, wait, or find other alternatives.

Sorry, it appears my question was worded poorly and came off as a criticism. I have huge respect for the work that the maintainers do, and I was not intending on rushing them.

What I had meant was to ask if I should remove Indium from my instance for now while the update is pending approval, or if I should use the jar provided in this PR while I wait instead. The first option is based on the assumption that the release process won't take a huge amount of time, so I can get away with no Indium for now, hence me asking if it'd come out soon.

CC: @sternschnaube

**Comment by D-argon** [comment]
> Sorry

nw

> What I had meant was to ask if I should remove Indium from my instance for now while the update is pending approval, or if I should use the jar provided in this PR 

as nobody seemed to have an grave error with Vieur version
I'd recommend you use it, as you'll either not have an error until the official release, so you wouldn't have to wait downgraded,
or have, so you can report and aid the fix
specially bc idk the project leader and his free time, and  if its soon or long, it's a wait

**Comment by sternschnaube** [comment]
I would follow try and error.

If everything works, then it's great, if not, report the bug and or downgrade to a stable setup from before 😊

**Comment by Vieur** [comment]
Thank you guys for positive feedbacks, as of now I am still currently learning java and stuff related to these kind of things! As for the bugs you may encounter, I am sorry because I only know the basics of porting some mods that are easy to port to newer version of Minecraft, but fixing bugs is still out of my knowledge. :(

**Comment by GODLIKE-ro** [comment]
needs a bump for sodium 0.5.5

File: gradle.properties
Comment (JonasFranke) [inline gradle.properties]: bump sodium to 0.5.5

**Comment by Eutropios** [comment]
> Thank you guys for positive feedbacks, as of now I am still currently learning java and stuff related to these kind of things! As for the bugs you may encounter, I am sorry because I only know the basics of porting some mods that are easy to port to newer version of Minecraft, but fixing bugs is still out of my knowledge. :(

Don't sweat it! You did a good job, and I'm sure a lot of us are appreciative of your work. Bugs are a normal part of software development.

**Comment by ronald69420** [comment]
Issue: there are lines going horizontal and vertical in between each glass block underwater

File: src/main/java/link/infra/indium/renderer/render/TerrainRenderContext.java
Comment (douira) [inline src/main/java/link/infra/indium/renderer/render/TerrainRenderContext.java:22]: The changes to this file are unnecessary since the imports aren't used and the other change was just whitesapce.

**Comment by D-argon** [comment]
> Issue: there are lines going horizontal and vertical in between each glass block underwater

if I reproduced the same issue, it is a vanilla problem apparently.
although it would help more info/screenshots

**Comment by douira** [comment]
Is this issue exclusive to Indium in this PR or just a general issue with the latest Sodium and/or vanilla?

**Comment by jokrefta** [comment]
Pretty sure thiis is a long standing issue w/ sodium. See for example, https://github.com/CaffeineMC/sodium-fabric/issues/38

**Comment by douira** [comment]
Ah well if you mean that then it's not a problem with this PR specifically. (I'm working on the translucency sorting issue atm)

**Comment by D-argon** [comment]
> Pretty sure thiis is a long standing issue w/ sodium. See for example, [CaffeineMC/sodium-fabric#38](https://github.com/CaffeineMC/sodium-fabric/issues/38)

this says that the water isn't shown through (at the other side of) glass, right?
The issue metioned is that there are transparent lines interrupting the rendering of the inner glass, in the block corners, underwater, and it happens at vannila, so it isnt sodium/indium related *apparently*

from what I found, it is related to the block outline when focused, as it gets less lines in F1

**Comment by douira** [comment]
The block outline for the focused block is a separate issue. What you are describing sounds like a lack of precision or intentional offsetting to prevent z fighting. I'm not quite sure what kind of situation you are seeing, a few screenshots would be helpful. But if it's not related to indium and happens in vanilla too this probably isn't the best place to discuss it.

**Pull request closed without merging**
